spec/classes/puppet_server_config_spec.rb

```diff
@@ -15,29 +15,31 @@
       }) end
 
       if Puppet.version < '4.0'
-        codedir          = '/etc/puppet'
-        conf_file        = '/etc/puppet/puppet.conf'
-        environments_dir = '/etc/puppet/environments'
-        logdir           = '/var/log/puppet'
-        rundir           = '/var/run/puppet'
-        vardir           = '/var/lib/puppet'
-        ssldir           = '/var/lib/puppet/ssl'
-        sharedir         = '/usr/share/puppet'
-        etcdir           = '/etc/puppet'
-        puppetcacmd      = '/usr/bin/puppet cert'
-        additional_facts = {}
+        codedir             = '/etc/puppet'
+        conf_file           = '/etc/puppet/puppet.conf'
+        environments_dir    = '/etc/puppet/environments'
+        logdir              = '/var/log/puppet'
+        rundir              = '/var/run/puppet'
+        vardir              = '/var/lib/puppet'
+        puppetserver_vardir = '/var/lib/puppet'
+        ssldir              = '/var/lib/puppet/ssl'
+        sharedir            = '/usr/share/puppet'
+        etcdir              = '/etc/puppet'
+        puppetcacmd         = '/usr/bin/puppet cert'
+        additional_facts    = {}
       else
-        codedir          = '/etc/puppetlabs/code'
-        conf_file        = '/etc/puppetlabs/puppet/puppet.conf'
-        environments_dir = '/etc/puppetlabs/code/environments'
-        logdir           = '/var/log/puppetlabs/puppet'
-        rundir           = '/var/run/puppetlabs'
-        vardir           = '/opt/puppetlabs/puppet/cache'
-        ssldir           = '/etc/puppetlabs/puppet/ssl'
-        sharedir         = '/opt/puppetlabs/puppet'
-        etcdir           = '/etc/puppetlabs/puppet'
-        puppetcacmd      = '/opt/puppetlabs/bin/puppet cert'
-        additional_facts = {:rubysitedir => '/opt/puppetlabs/puppet/lib/ruby/site_ruby/2.1.0'}
+        codedir             = '/etc/puppetlabs/code'
+        conf_file           = '/etc/puppetlabs/puppet/puppet.conf'
+        environments_dir    = '/etc/puppetlabs/code/environments'
+        logdir              = '/var/log/puppetlabs/puppet'
+        rundir              = '/var/run/puppetlabs'
+        vardir              = '/opt/puppetlabs/puppet/cache'
+        puppetserver_vardir = '/opt/puppetlabs/server/data/puppetserver'
+        ssldir              = '/etc/puppetlabs/puppet/ssl'
+        sharedir            = '/opt/puppetlabs/puppet'
+        etcdir              = '/etc/puppetlabs/puppet'
+        puppetcacmd         = '/opt/puppetlabs/bin/puppet cert'
+        additional_facts    = {:rubysitedir => '/opt/puppetlabs/puppet/lib/ruby/site_ruby/2.1.0'}
       end
 
       if os_facts[:osfamily] == 'FreeBSD'
```

Comment:
I think you need to set `puppetserver_vardir` here too, as there are some test failures:

<pre>
  1) puppet::server::config on freebsd-10-amd64 with no custom parameters should set up the ENC

     Failure/Error: })

       expected that the catalogue would contain Class[foreman::puppetmaster] with puppet_home set to "/var/lib/puppet" but it is set to "/var/puppet"
</pre>